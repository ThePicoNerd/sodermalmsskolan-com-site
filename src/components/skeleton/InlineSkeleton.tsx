import React, { FunctionComponent } from "react";
import Skeleton, { SkeletonProps } from "./Skeleton";

export interface InlineSkeletonProps extends SkeletonProps {
  count?: number;
}

const InlineSkeleton: FunctionComponent<InlineSkeletonProps> = ({
  count = 1,
  ...skeletonProps
}) => (
  <span>
    {Array.from({ length: count }).map((_, index) => (
      // eslint-disable-next-line react/no-array-index-key
      <Skeleton key={index} {...skeletonProps}>
        &zwnj;
      </Skeleton>
    ))}
  </span>
);

export default InlineSkeleton;
