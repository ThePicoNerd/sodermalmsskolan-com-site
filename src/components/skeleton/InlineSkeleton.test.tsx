import React from "react";
import { render } from "@testing-library/react";
import InlineSkeleton from "./InlineSkeleton";

test("<InlineSkeleton />", () => {
  const result = render(<InlineSkeleton />);

  expect(result.asFragment()).toMatchSnapshot();
});
