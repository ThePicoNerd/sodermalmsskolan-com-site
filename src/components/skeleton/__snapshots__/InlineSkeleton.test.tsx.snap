// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineSkeleton /> 1`] = `
<DocumentFragment>
  <span>
    <span
      class="skeleton max-w-full"
    >
      â€Œ
    </span>
  </span>
</DocumentFragment>
`;
